README.md

```diff
@@ -1,33 +1,230 @@
 
+While the main SwearGame contract conduct a trial to resolve a specific case it iterate between different
+trial statuses.
 
+For each game there is a specific pre defined trial statuses and a specific trial transitions scheme which defined
+the transition for a certain status to the next one.
 
+The trial statuses and their transitions are defined per each game in a TrialRules contract.
 
+The TrialRules contract also defined the grace period per each trial status.
 
+This TrialRules contract is align with TrialRulesAbstract.sol  and should implement the following ABIs:
+
+/// @notice getStatus - get next trial status according to witness state and the current trial state
+
+/// @param witnessStatus witness status (VALID , INVALID,PENDING)
+
+/// @param trialStatus current trial status
+
+/// @return status - next trial status - can be also GUILTY or NOT GUILTY.
+
+function getStatus(uint8 witnessStatus,uint8 trialStatus) returns (uint8 status);
+
+/// @notice getWitness - get witness according to the trial status
+
+/// @param trialStatus current trial status
+
+/// @return WitnessAbstract - return a witness contract instance
+
+function getWitness(uint8 trialStatus) returns (WitnessAbstract);
+
+/// @notice getInitialStatus - get initial trial status
+
+/// @return status -
+
+function getInitialStatus() public returns (uint8 status);
+
+/// @notice expired - check expiration for a certain case and trial status.
+
+/// @return bool - true if expired otherwise false.
+
+function expired(bytes32 caseId,uint8 status) returns (bool);
+
+/// @notice startGracePeriod - start counting for a grace period for a certain case and status.
+
+/// @return bool - true if it actually start counting for the grace period
+
+///                false -if the grace period already started
+
+function startGracePeriod(bytes32 caseId,uint8 status) returns (bool);
+
+/// @notice getReward - return the reward for a valid case
+
+///
+
+/// @return reward - the reward for a valid case
+
+function getReward()returns (uint reward);
+
+
+
+## Mirror game
+A very simple basic game where the service offers to mirror the client ENS("client.game") content on its own ENS("service.game").
+
+### game flow
+-service will deposit some funds in the swear contact to ensure the service in the case of a court case/litigation
+
+-client receive a signed promise from the service to mirror its ens during the next X blocks.
+
+-client update the client.game ENS.
+
+If after X blocks the service ENS does not "mirror" its client ENS ..the client can submit a new case for the swear contract.
+
+As an evidence the client submits to court the signed promise it got from the service and the ENSNameHashes for both client and service.
+
+If the case is valid a refund + compensation will send to the client from the contract.
+
+### witnesses
+ - PromiseValidator - validate a signed promise for the case it is submitted and it is ask for testimony .
```

Comment:
Not sure if i understand this. Is the below correct?

validates a signed promise for the case it is submitted to and asks for a testimony.